lib/Jsrt/Core/JsrtContextCore.h

```diff
@@ -259,6 +265,12 @@ class ChakraCoreHostScriptContext sealed : public HostScriptContext
     void SetFetchImportedModuleFromScriptCallback(FetchImportedModuleFromScriptCallBack fetchCallback) { this->fetchImportedModuleFromScriptCallback = fetchCallback; }
     FetchImportedModuleFromScriptCallBack GetFetchImportedModuleFromScriptCallback() const { return this->fetchImportedModuleFromScriptCallback; }
 
+    void SetHostFinalizeImportMetaCallback(HostFinalizeImportMetaCallback finalizeCallback) { this->hostFinalizeImportMetaCallback = finalizeCallback; }
+    HostFinalizeImportMetaCallback GetHostFinalizeImportMetaCallback() const { return this->hostFinalizeImportMetaCallback; }
+
+    void SetHostGetImportMetaPropertiesCallback(HostGetImportMetaPropertiesCallback getCallback) { this->hostGetImportMetaPropertiesCallback = getCallback; }
```

Comment:
We're doing something just a bit different from the spec-level operations here, in the sense that the spec's `HostGetImportMetaProperties` takes a module record and returns a (spec-internal) "List" of property records, whereas here we are giving the embedder an object and letting them do what they want with it.

Similarly, since we are giving the embedder the object and they can do anything they want with it, (e.g. freeze it or anything else) there's probably no advantage in having a separate "finalize" callback.

V8 chose to have just one callback ("HostInitializeImportMetaObjectCallback"). What do you think about just having one ourselves?

With respect to naming, I think either of these would be fine (I don't think we're necessarily obligated to use the "Host*" naming convention):

- `JsSetInitializeImportMetaCallback`
- `JsSetHostInitializeImportMetaCallback`

What do you think?

Reply:
Yes, should have called this change out in the PR description. 

I took the liberty of changing `HostGetImportMetaProperties` because it seemed to be more work than it was worth trying to build some way for the host API to return us a `List` of key/value pairs we would then set as properties in the import.meta object. The way we're supposed to give the host the import.meta object anyway via the finalize callback means we aren't using the `List` to avoid handing out the object itself to the host.

I knew V8 had implemented the callback this way which is why I initially only wrote one callback for ChakraCore as well. I added the other (and kept the spec proposal names) in an attempt to hew as closely to the spec as makes sense. 

However, I agree completely that having both callbacks (which have the same signature and arguments) and calling them back to back does look a little silly. I have no problem coalescing the two into a single API with a more straight-forward name. Probably prefer `InitializeImportMetaCallback`. The reason I wrote it this way up-front was to keep it spec-focused and have a conversation like this here in the comments.